File: python/cart_shipment_quotes.py
import sdk.python.entities
from sdk.python.entities import Property


class CartShipmentQuote(sdk.python.entities.Entity):

    resource = '/cart_shipment_quotes/'
    json_name = 'cartShipmentQuote'

    id = Property(int)
    name = Property(str)
    subtotal_cost = Property(float)
    tax_amount = Property(float)
    total_cost = Property(float)
    shipment_method = Property("sdk.python.shipment_methods.ShipmentMethod")
    shipment_service = Property(int)
    shipment_service_quote = Property(str)


class CartShipmentQuotes(sdk.python.entities.Resource):

    entity_class = CartShipmentQuote
    json_name = 'cartShipmentQuotes'


cart_shipment_quotes = CartShipmentQuotes()

File: php/src/cart_shipment_quotes.php
<?php

declare(strict_types=1);

require_once 'entity.php';

class CartShipmentQuote extends Entity
{
    public static $json_name = 'cart_shipment_quote';
    public static $resource = '/cart_shipment_quotes/';


    public function __construct()
    {
        $this->json_property('id', 'integer');
        $this->json_property('name', 'string');
        $this->json_property('subtotalCost', 'integer');
        $this->json_property('taxAmount', 'integer');
        $this->json_property('totalCost', 'integer');
        $this->json_property('shipmentServiceQuote', 'string');


        $this->json_property('shipmentService', 'ShipmentService', $many = True,
            $recursive = True);
        $this->json_property('shipmentMethod', 'ShipmentMethod', $many = True,
            $recursive = True);
    }
}
